core/lib/Cardano/Address/Script.hs

```diff
@@ -252,25 +372,59 @@ keyHashFromBytes (cred, bytes)
     | otherwise = Just $ KeyHash cred bytes
 
 -- | Encode a 'KeyHash' to bech32 'Text' or hex is key role unknown.
+--  If one wants to include, valid in governance roles only, additional byte
+--  as specified in CIP-0129, the function needs to be called with withByte=true.
 --
 -- @since 3.0.0
-keyHashToText :: KeyHash -> Text
-keyHashToText (KeyHash cred keyHash) = case cred of
+keyHashToText :: KeyHash -> Bool -> Text
+keyHashToText (KeyHash cred keyHash) withByte = case cred of
     Payment ->
         T.decodeUtf8 $ encode (EBech32 CIP5.addr_shared_vkh) keyHash
     Delegation ->
         T.decodeUtf8 $ encode (EBech32 CIP5.stake_shared_vkh) keyHash
     Policy ->
         T.decodeUtf8 $ encode (EBech32 CIP5.policy_vkh) keyHash
     Representative ->
-        T.decodeUtf8 $ encode (EBech32 CIP5.drep) keyHash
+        if withByte then
+            T.decodeUtf8 $ encode (EBech32 CIP5.drep) $ keyHashAppendByteCIP0129 keyHash cred
+        else
+            T.decodeUtf8 $ encode (EBech32 CIP5.drep_vkh) keyHash
     CommitteeCold ->
-        T.decodeUtf8 $ encode (EBech32 CIP5.cc_cold) keyHash
+        if withByte then
+            T.decodeUtf8 $ encode (EBech32 CIP5.cc_cold) $ keyHashAppendByteCIP0129 keyHash cred
+        else
+            T.decodeUtf8 $ encode (EBech32 CIP5.cc_cold_vkh) keyHash
     CommitteeHot ->
-        T.decodeUtf8 $ encode (EBech32 CIP5.cc_hot) keyHash
+        if withByte then
+            T.decodeUtf8 $ encode (EBech32 CIP5.cc_hot) $ keyHashAppendByteCIP0129 keyHash cred
+        else
+            T.decodeUtf8 $ encode (EBech32 CIP5.cc_hot_vkh) keyHash
     Unknown ->
         T.decodeUtf8 $ encode EBase16 keyHash
 
+-- | In accordance to CIP-0129 (https://github.com/cardano-foundation/CIPs/tree/master/CIP-0129)
+--   one byte is prepended to vkh only in governance context. The rules how to contruct it are summarized
+--   below
+--
+--   drep       0010....
+--   hot        0000....    key type
+--   cold       0001....
+--
+--   keyhash    ....0010
+--   This is on top of X_vkh, where X={drep, cc_hot, cc_hot}, which lacks the additional byte.
+--   In `keyHashFromText` we additionally
+--   support reading legacy X which also lacks the additional byte, and has the same payload as
+--   as the corresponding X_vkh.
+keyHashAppendByteCIP0129 :: ByteString -> KeyRole -> ByteString
```

Comment:
This is the same comment than in the `scriptToHashText` function, perhaps just a link would suffice?